### PR TITLE
Add a nix flake

### PR DESCRIPTION
**Pull request by nixos-goddess**
So, I switched back to nixos, which means that a flake would make my life much easier when coding magmawm. It would also do other potential contributors and users who use nix a big favor.

### CODE REVIEW

**Comment by electron271** [comment]
resolves #46

**Comment by nixos-goddess** [comment]
well, it doesn't really resolve #46. As of now, it only contains the instructions to make an isolated development environment that contains the appropriate rust toolchain and dependencies in order to be able to compile. But, with a flake, we can easily add nix modules.

**Comment by nixos-goddess** [comment]
I am not sure if nix has a function to convert nix into ron tho. I'll have to check nixpkgs and the language docs

**Comment by electron271** [comment]
out of curiosity why not just.. use the config file

**Comment by nixos-goddess** [comment]
If we can do a lot with magmawm in nix, including support for nix-friendly config, it'll make for a good declarative environment. Nix users love declarative environments :p. If we have good nix modularity, we can become the goto wm for nixos.

**Comment by nixos-goddess** [comment]
made a package :3

File: flake.nix [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+            pname = "magmawm";
+	    version = "0.0.1";
+	    src = ./.;
+	    buildInputs = dependencies;
+	    nativeBuildInputs = native-dependencies;
+
+	    cargoLock = {
+              lockFile = ./Cargo.lock;
+	      outputHashes = {
+                "smithay-0.3.0" = "sha256-UQwYV3GOkZHEFyAMRMY46JiFXsdsjfSVHAHlA47ZAtI=";
```
Comment (electron271) [inline flake.nix:60]: We are not using the release version of smithay, we use the git version.

**Review by electron271 — CHANGES_REQUESTED** [review]

File: flake.nix (hunk 1 above)
Comment (nixos-goddess) [reply flake.nix:60]: I am aware. The version number i put was the crate version number, not the github release version number.

Comment (electron271) [reply flake.nix:60]: we use the git crate though, im unsure what you mean here

```toml
[dependencies.smithay]
git = "https://github.com/Smithay/smithay.git"
default-features = false
features = [
    "backend_winit",
    "wayland_frontend",
    "desktop",
    "backend_session_libseat",
    "backend_drm",
    "renderer_multi",
    "renderer_glow",
    "backend_gbm",
    "backend_udev",
    "backend_egl",
    "backend_libinput",
    "use_system_lib",
]

```

Comment (nixos-goddess) [reply flake.nix:60]: We do use the git crate, but crates have a version field in their Cargo.toml field. On smithay's github, the version field for the cargo project is 0.3.0

**Review by electron271 — APPROVED** [review]
looks okay, rather if someone else review first before merging as i dont use nix

@Insert5StarName please review

**Review by electron271 — APPROVED** [review]
gonna face it, amy disappeared

force merging basically

**Pull request merged**